### PR TITLE
Add Github Actions Testing to Examples

### PR DESCRIPTION
**Pull request by boray**
This PR updates the test-tutorials workflow to run examples when a commit is pushed to the main branch.

- Bump node version to 20
- Added automated testing for the following examples:
- `01-hello-world`
- `02-private-inputs-and-hash-functions`
- `05-common-types-and-functions`
- `10-account-updates`

### CODE REVIEW

File: .github/workflows/test-tutorials.yml [hunk 1]
```diff
@@ -2,18 +2,41 @@ name: Test-Tutorials
...
+      - name: Install dependencies at root level
+        run: npm ci
+      
+      - name: Set Git config (global)
+        run: |
           git config --global user.email "test@example.com"
           git config --global user.name "Test"
-          npx ts-node scripts/tutorial-runner.ts docs/zkapps/tutorials/01-hello-world.mdx
+
+      - name: Install and run build for ${{ matrix.example-path }}
```
Comment (shimkiv) [inline .github/workflows/test-tutorials.yml:37]:
```suggestion
      - name: Install dependencies, build and run ${{ matrix.example-path }}/build/src/main.js
```

File: .github/workflows/test-tutorials.yml [hunk 2]
```diff
@@ -2,18 +2,41 @@ name: Test-Tutorials
...
         with:
-          # In case of version change, update README.md accordingly
-          node-version: 18
-      - run: |
-          npm ci
+          node-version: '20'
+      
+      - name: Install dependencies at root level
+        run: npm ci
+
```
Comment (shimkiv) [inline .github/workflows/test-tutorials.yml:31]:
```suggestion
```

File: .github/workflows/test-tutorials.yml [hunk 3]
```diff
@@ -2,18 +2,41 @@ name: Test-Tutorials
...
+          - examples/zkapps/01-hello-world
+          - examples/zkapps/02-private-inputs-and-hash-functions
+          - examples/zkapps/05-common-types-and-functions
+          - examples/zkapps/10-account-updates
+
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v3
+      - name: Checkout repository
+        uses: actions/checkout@v2
```
Comment (shimkiv) [inline .github/workflows/test-tutorials.yml:22]:
```suggestion
        uses: actions/checkout@v4
```

File: .github/workflows/test-tutorials.yml [hunk 4]
```diff
@@ -2,18 +2,41 @@ name: Test-Tutorials
...
+      
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
```
Comment (shimkiv) [inline .github/workflows/test-tutorials.yml:25]:
```suggestion
        uses: actions/setup-node@v4
```

File: .github/workflows/test-tutorials.yml [hunk 5]
```diff
@@ -2,18 +2,41 @@ name: Test-Tutorials
 
 on:
   push:
+    branches:
+      - main
 
 jobs:
-  hello-world:
+  run-tests:
```
Comment (shimkiv) [inline .github/workflows/test-tutorials.yml:9]:
```suggestion
  test-examples:
```

File: .github/workflows/test-tutorials.yml [hunk 6]
```diff
@@ -2,18 +2,38 @@ name: Test-Tutorials
...
+        uses: actions/checkout@v4
+      
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
         with:
-          # In case of version change, update README.md accordingly
-          node-version: 18
-      - run: |
-          npm ci
+          node-version: '20'
```
Comment (45930) [inline .github/workflows/test-tutorials.yml:27]: With the matrix strategy, we can run this on node 18, 20 and 22.  I think we might as well, so we know what works and doesn't work.

File: .github/workflows/test-tutorials.yml [hunk 7]
```diff
@@ -2,18 +2,38 @@ name: Test-Tutorials
 
 on:
   push:
+    branches:
+      - main
```
Comment (shimkiv) [inline .github/workflows/test-tutorials.yml:6]:
```suggestion
on:
  push:
    branches:
      - main
  workflow_dispatch: {}
```

**Review by 45930 — APPROVED** [review]
This is a good step!  Thanks for implementing it.  I left one suggestion about using multiple node versions for broader coverage.

File: .github/workflows/test-tutorials.yml (hunk 6 above)
Comment (boray) [reply .github/workflows/test-tutorials.yml:29]: It would be good to check, but do you think it’s worth the 3x time cost?

Comment (shimkiv) [reply .github/workflows/test-tutorials.yml:29]: Jobs that use matrix will be executed in parallel, so no extra cost.

File: .github/workflows/test-tutorials.yml (hunk 7 above)
Comment (boray) [reply .github/workflows/test-tutorials.yml:7]: There was already a `name: Test-Tutorials` in the first line. I'll remove the one added with this suggestion. Does it work for you?

Comment (boray) [reply .github/workflows/test-tutorials.yml:5]: Also, run workflow doesn't show up in Actions. Are you able to run it?

Comment (shimkiv) [reply .github/workflows/test-tutorials.yml:5]: These changes have to be merged into the `main` first to let the workflow be manually triggered.

**Review by shimkiv — APPROVED** [review]

File: .github/workflows/test-tutorials.yml (hunk 7 above)
Comment (shimkiv) [reply .github/workflows/test-tutorials.yml:5]: If you want to make this workflow to execute on every push and PR then it should be something like:

```yaml
on:
  push: {}
  pull_request: {}
  workflow_dispatch: {}
```

Comment (boray) [reply .github/workflows/test-tutorials.yml:5]: I don’t think it should be executed on every push and PR, as changes in doc2 won’t affect how examples run. Running it on every push to main is sufficient, IMO.

**Pull request merged**